hibernate6-jpa/src/main/java/io/micronaut/configuration/hibernate6/jpa/graal/HibernateSubstitutions.java

```diff
@@ -138,6 +143,19 @@ final class Hql {
 final class Cfg {
 }
 
+// Disable Runtime Byte Code Enhancement
+@TargetClass(className = "org.hibernate.cfg.Environment")
+@TypeHint(
+    value = {EventType.class, EnumType.class},
+    accessType = {AccessType.ALL_DECLARED_FIELDS, AccessType.ALL_DECLARED_METHODS, AccessType.ALL_DECLARED_CONSTRUCTORS}
+)
+final class EnvironmentSubs {
+    @Substitute
+    public static BytecodeProvider buildBytecodeProvider(Properties properties) {
+        return new IntrospectedHibernateBytecodeProvider();
```

Comment:
Added back since now it is possible to inject custom bytecode provider (hibernate 6.2).